nonarch.imp

```diff
@@ -0,0 +1,19 @@
+[
+    # If the directory is not architecture specific, add this filter to avoid
+    # excess asm inclusions
+    { include: ["@.asm/ptrace.h.", private, "<linux/ptrace.h>", "public"] }, 
+    { include: ["@.asm/bug.h.", private, "<linux/bug.h>", "public"] },
+    { include: ["@.asm/delay.h.", private, "<linux/delay.h>", "public"] },
+    { include: ["@.asm/io.h.", private, "<linux/io.h>", "public"] },
+    { include: ["@.asm/kprobes.h.", private, "<linux/kprobes.h>", "public"] },
```

Comment:
I'm still struggling with the distinction of "arch specific" vs not.

Consider this rule for kprobes.h.  You have it in this table, so it sounds like it's "not architecture specific."  But I don't see instances of it for:
- alpha
- microblaze
- nios2
- openrisc
- xtensa

So it's still not clear to me; let's say I need to add a rule for an asm/*.h -> linux/*.h mapping.  When should I add it to nonarch.imp vs filter.imp.  That's still unclear to me.  Saying "well is it architecture specific or not" doesn't clarify; what is your definition of "architecture specific?"